biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -1007,6 +1007,75 @@ public String getBsn() throws Exception {
 		return null;
 	}
 
+	/**
+	 * Guess the module name of the bundle from
+	 * <p>
+	 * a) the module-info.class or,
+	 * <p>
+	 * b) the {@link Constants#AUTOMATIC_MODULE_NAME} manifest header or,
+	 * <p>
+	 * c) the {@link Constants#BUNDLE_SYMBOLICNAME} manifest header.
+	 *
+	 * @return null when the jar has no module-info.class, manifest, when the
+	 *         manifest has no {@link Constants#AUTOMATIC_MODULE_NAME} header
+	 *         and no {@link Constants#BUNDLE_SYMBOLICNAME} header, or when
+	 *         either header value is not a valid bsn according to {@link #BSN}.
+	 * @throws Exception when the jar is closed or when the manifest could not
+	 *             be retrieved.
+	 */
+	public String getModuleName(Analyzer analyzer) throws Exception {
+		check();
+		Resource resource = getResource(Constants.MODULE_INFO_CLASS);
```

Comment:
Do we want to create a new Clazz here? Hmm, this seems odd since you need to pass an Analyzer in to this method. This is the first time Jar touches Analyzer which seems inverted.

I am not sure what the right solution here is but this feels odd. I need to think more. It also most seems like Analyzer needs a method which returns the module name of the bundle being analyzed.